cmd/run.go

```diff
@@ -445,18 +449,44 @@ a commandline interface for interacting with it.`,
 			logrus.Warn("No data generated, because no script iterations finished, consider making the test duration longer")
 		}
 
+		data := ui.SummaryData{
+			Opts:    conf.Options,
+			Root:    engine.Executor.GetRunner().GetDefaultGroup(),
+			Metrics: engine.Metrics,
+			Time:    engine.Executor.GetTime(),
+		}
+
 		// Print the end-of-test summary.
 		if !conf.NoSummary.Bool {
 			fprintf(stdout, "\n")
-			ui.Summarize(stdout, "", ui.SummaryData{
-				Opts:    conf.Options,
-				Root:    engine.Executor.GetRunner().GetDefaultGroup(),
-				Metrics: engine.Metrics,
-				Time:    engine.Executor.GetTime(),
-			})
+			ui.Summarize(stdout, "", data)
 			fprintf(stdout, "\n")
 		}
 
+		if conf.SummaryExport.Valid {
+			var w io.Writer
+			switch conf.SummaryExport.String {
+			case "": // disable, do nothing
+			case "/dev/stdout":
+				w = stdout
+			default:
+				f, err := os.Create(conf.SummaryExport.String)
+				if err != nil {
+					logrus.Error("failed to create report file")
+					return err
+				}
+				w = f
+				defer func() {
+					if err := f.Close(); err != nil {
+						logrus.WithError(err).Fatal("failed to close summary output file")
+					}
+				}()
+			}
+			if w != nil {
+				ui.SummarizeJSON(w, data)
+			}
+		}
+
```

Comment:
Again, having a way to output to stdout in anyway currently will be *useless* so here you should only care if the string is not empty and open a file and write to it, nothing more, nothing less. 

Also exiting from this function to return this error means the code below won't be executed, which in my opinion is important to execute, specifically users will probably be interested in the fact that thresholds have failed ...

Reply:
@MStoykov as discussion with @na--  yesterday, we still support dumping to stdout, unless something changes that I did not notice.

Reply:
@MStoykov fixed after re-ask @na--